core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -201,8 +205,8 @@ class SYCLInternal {
   // An indirect kernel is one where the functor to be executed is explicitly
   // copied to USM memory before being executed, to get around the
   // trivially copyable limitation of SYCL.
-  using IndirectKernelMem = USMObjectMem<sycl::usm::alloc::shared>;
-  IndirectKernelMem m_indirectKernelMem;
+  using IndirectKernelMem = USMObjectMem<sycl::usm::alloc::device>;
+  IndirectKernelMem& get_indirect_kernel_mem();
 
   using IndirectReducerMem = USMObjectMem<sycl::usm::alloc::shared>;
   IndirectReducerMem m_indirectReducerMem;
```

Comment:
Let's just get rid of this member variable when we are introducing a pool.

Reply:
Done - using the single USM pool for both.